### PR TITLE
Add some new features

### PR DESCRIPTION
**Pull request by ahadvl**
In this update, next js updated to latest version, view button of mushaf and surah in table moved and used new delete button and it will be work correctly in next update soon.

### CODE REVIEW

File: app/panel/mushaf/list/page.tsx [hunk 1]
```diff
@@ -36,8 +37,8 @@ export default async function Page() {
                     <Button variant="outlined">Add Mushaf</Button>
                 </Link>
             </Row>
-            <Table>
-                <Thead style={{ textAlign: "left" }}>
+            <Table style={{ textAlign: "justify" }}>
+                <Thead>
```
Comment (Al-Abd) [inline app/panel/mushaf/list/page.tsx:41]: use justify style on thead

File: app/panel/mushaf/list/page.tsx [hunk 2]
```diff
@@ -48,7 +49,11 @@ export default async function Page() {
                 <Tbody>
                     {mushafsList.map((item) => (
                         <Tr>
-                            <Td>{item.name}</Td>
+                            <Td>
+                                <Link href={"/panel/mushaf/" + item.uuid}>
```
Comment (Al-Abd) [inline app/panel/mushaf/list/page.tsx:53]: better to remove this link, it must affect on tr, currently remove it.

File: app/panel/mushaf/list/page.tsx [hunk 3]
```diff
@@ -58,21 +63,17 @@ export default async function Page() {
...
-                                    </Link>
                                     <Link
                                         href={"/panel/mushaf/edit/" + item.uuid}
                                     >
                                         <Button size="small" variant="link">
                                             Edit
                                         </Button>
                                     </Link>
-                                    <Button size="small" variant="link">
+                                    <DeleteButton />
```
Comment (Al-Abd) [inline app/panel/mushaf/list/page.tsx:73]: `<deleteButton controller="mushaf" uuid={item.uuid} />`

File: app/panel/surah/list/page.tsx [hunk 4]
```diff
@@ -31,8 +32,8 @@ export default async function Page() {
     return (
         <Container>
             <h1>Surah List</h1>
-            <Table>
-                <Thead style={{ textAlign: "left" }}>
+            <Table style={{ textAlign: "justify" }}>
+                <Thead>
```
Comment (Al-Abd) [inline app/panel/surah/list/page.tsx:36]: style

File: app/panel/surah/list/page.tsx [hunk 5]
```diff
@@ -44,27 +45,29 @@ export default async function Page() {
                 <Tbody>
                     {surahsList.map((item) => (
                         <Tr>
-                            <Td>{item.number}</Td>
-                            <Td>{item.name}</Td>
+                            <Td>
+                                <Link href={"/panel/surah/" + item.uuid}>
```
Comment (Al-Abd) [inline app/panel/surah/list/page.tsx:49]: remove links from td inside and revive view button

File: app/panel/surah/list/page.tsx [hunk 6]
```diff
@@ -44,27 +45,29 @@ export default async function Page() {
...
                                         href={"/panel/surah/edit/" + item.uuid}
                                     >
                                         <Button size="small" variant="link">
                                             Edit
                                         </Button>
                                     </Link>
-                                    <Button size="small" variant="link">
-                                        Delete
-                                    </Button>
+                                    <DeleteButton />
```
Comment (Al-Abd) [inline app/panel/surah/list/page.tsx:70]: `<deleteButton controller="mushaf" uuid={item.uuid} />`

File: components/deleteButton.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { Button, Container, Form, InputField, Row, Spacer } from "@yakad/ui";
+import { useRouter } from "next/navigation";
+import { useFetch, useFormDataHandle } from "@yakad/lib";
+import React from "react";
```
Comment (Al-Abd) [inline components/deleteButton.tsx:6]: importing order:

ract
next/nav
yakad/lib
yakad/ui

File: components/deleteButton.tsx [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+interface Delete {
```
Comment (Al-Abd) [inline components/deleteButton.tsx:8]: `interface DeleteButtonProps extends React.HTMLAttributes<HTMLButtonElement> {`

File: components/deleteButton.tsx [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    controller: string;
+    uuid: string;
+}
+
+export default function DeleteButton() {
```
Comment (Al-Abd) [inline components/deleteButton.tsx:13]: export default function DeleteButton(props: DeleteButtonProps) {

File: components/deleteButton.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
+"use client";
+
+import { Button, Container, Form, InputField, Row, Spacer } from "@yakad/ui";
+import { useRouter } from "next/navigation";
+import { useFetch, useFormDataHandle } from "@yakad/lib";
+import React from "react";
+
+interface Delete {
+    controller: string;
+    uuid: string;
```
Comment (Al-Abd) [inline components/deleteButton.tsx:10]: add `?name: string` to interface

File: components/deleteButton.tsx [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+                Accept: "application/json",
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(formData),
+        }
+    );
+
+    const handler = useFormDataHandle(setFormData);
+    function deleted() {
+        if (confirm("Are you sure to delete a uuid") == true) {
```
Comment (Al-Abd) [inline components/deleteButton.tsx:31]: `if (confirm("Are you sure to delete UUID:" + props.uuid) == true) {`

or:

`if (confirm("Are you sure to delete: " + props.name ? props.name : props.uuid) == true) {`

File: components/deleteButton.tsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+import React from "react";
+
+interface Delete {
+    controller: string;
+    uuid: string;
+}
+
+export default function DeleteButton() {
+    const router = useRouter();
+    const [formData, setFormData] = React.useState<Delete>();
```
Comment (Al-Abd) [inline components/deleteButton.tsx:15]: i think we dont have form and we must remove this

File: components/deleteButton.tsx [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+import { useFetch, useFormDataHandle } from "@yakad/lib";
+import React from "react";
+
+interface Delete {
+    controller: string;
+    uuid: string;
+}
+
+export default function DeleteButton() {
+    const router = useRouter();
```
Comment (Al-Abd) [inline components/deleteButton.tsx:14]: router is not used

File: components/deleteButton.tsx [hunk 14]
```diff
@@ -0,0 +1,69 @@
...
+    const [formData, setFormData] = React.useState<Delete>();
+
+    const fetch = useFetch(
+        `${process.env.NEXT_PUBLIC_API_URL}/?{controller/?{uuid}`,
```
Comment (Al-Abd) [inline components/deleteButton.tsx:18]: something like this:
`${process.env.NEXT_PUBLIC_API_URL}/?{props.controller/?{props.uuid}`

File: components/deleteButton.tsx [hunk 15]
```diff
@@ -0,0 +1,69 @@
...
+        {
+            method: "DELETE",
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
```
Comment (Al-Abd) [inline components/deleteButton.tsx:23]: We dont send json in delete method, so i think we can remove entire header, @AzerothA please answer this.

File: components/deleteButton.tsx [hunk 16]
```diff
@@ -0,0 +1,69 @@
...
+            },
+            body: JSON.stringify(formData),
```
Comment (Al-Abd) [inline components/deleteButton.tsx:25]: I think we can remove this too, @AzerothA .

File: components/deleteButton.tsx [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+        }
+    );
+
+    const handler = useFormDataHandle(setFormData);
+    function deleted() {
```
Comment (Al-Abd) [inline components/deleteButton.tsx:30]: change function name to: delete

File: components/deleteButton.tsx [hunk 18]
```diff
@@ -0,0 +1,69 @@
...
+            method: "DELETE",
+            headers: {
+                Accept: "application/json",
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify(formData),
+        }
+    );
+
+    const handler = useFormDataHandle(setFormData);
```
Comment (Al-Abd) [inline components/deleteButton.tsx:29]: remove handler too

File: components/deleteButton.tsx [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+        //         <p>The name of controller like mushaf, suarah & ...</p>
+        //         <InputField
+        //             variant="outlined"
+        //             placeholder="UUID"
+        //             type="string"
+        //             name="uuid"
+        //         />
+        //         <p>The mushaf, surah or ... uuid</p>
+        //     </Form>
+        // </Container>
```
Comment (Al-Abd) [inline components/deleteButton.tsx:59]: remove all these comments

File: components/deleteButton.tsx [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+        <Button
+            onClick={deleted}
+            size="small"
+            loadingVariant="spinner"
+            variant="link"
+        >
```
Comment (Al-Abd) [inline components/deleteButton.tsx:65]: we must replace line 62, 63, 62 with `{...props}` then set size & variant where you use deleteButton, currently you can't do this due to type errors, so don't do this one and make this as issue,  we must change yakad/ui and export yakad components type. @ekiuse .

File: package.json [hunk 21]
```diff
@@ -16,8 +16,8 @@
         "@yakad/symbols": "1.0.0",
         "@yakad/ui": "1.2.79-beta",
         "@yakad/x": "1.0.23-beta",
-        "eslint-config-next": "13.4.19",
-        "next": "13.4.19",
+        "eslint-config-next": "13.5.3",
+        "next": "13.5.3",
```
Comment (Al-Abd) [inline package.json:20]: don't update next every time.

**Review by Al-Abd — CHANGES_REQUESTED** [review]
Thank you.

**Comment by Al-Abd** [comment]
Fix problems in next pr.

File: components/deleteButton.tsx (hunk 15 above)
Comment (AzerothA) [reply components/deleteButton.tsx:23]: Yes, we should remove the entire header.

File: components/deleteButton.tsx (hunk 16 above)
Comment (AzerothA) [reply components/deleteButton.tsx:25]: Yes, the delete router accepts no HTTP body from the request.

**Pull request merged**